Allow specifying `LIBXML_` flags when sending/receiving/loading requests/responses

frederikbosch opened the pull request:
Supersedes #4.

Rationale: no member properties or state required.

## Client

Easy.

```php
$client->doRequest($request, null, LIBXML_PARSEHUGE);
```

## Server

Thought about changing `Http` and `Stdin` classes, decided not to. Let's leave those alone. Decided developer has to instantiate Request and then call `loadXml()`.

```php
$request = new \Laminas\XmlRpc\Request();
$request->loadXml(file_get_contents('php://input'), LIBXML_PARSEHUGE);
echo $server->handle($request);
```

frederikbosch committed:
add options to use libxml in server and client

Signed-off-by: Frederik Bosch <f.bosch@genkgo.nl>

weierophinney committed:
docs: document ability to use LIBXML constants

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

weierophinney reviewed (approved):
I like this!

I've added docs; otherwise no changes. :ship:

Ocramius commented:
@weierophinney I changed base branch - thanks for the review: I wasn't sure about this overall :)

weierophinney commented:
I've rebased against 2.13.x now, @Ocramius

Ocramius commented:
@weierophinney ok, overall fine with the patch then (and I now understand it, considering the added docs), but the docblocks should really have examples then @_@

weierophinney committed:
docs: provide docblock information on LIBXML options

Adds both a `@param` annotation for the `$libXmlOptions` arguments, and narrative information in the method docblocks describing it and where to find a list of options.
Omitted from non-public methods, however.

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

weierophinney commented:
> the docblocks should really have examples then

Done. :smile:

Ocramius reviewed (approved):
Excellent, thanks!

frederikbosch commented:
Thanks @Ocramius and @weierophinney
